tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -111,6 +111,12 @@ public int Serialize(ref byte[] bytes, int offset, in TraceChunkModel traceChunk
             // serialize each span
             for (var i = 0; i < traceChunk.SpanCount; i++)
             {
+                if (maxSize != null && offset - originalOffset >= maxSize)
```

Comment:
Would it make more sense to move this check to after the `Serialize(ref bytes, offset, in spanModel);`? In its current form, the very last span serialization could go over the limit and we would not detect it.

Reply:
I don't think it would help. Once the span has been serialized, the harm has been done.